compiler/rustc_ast/src/mut_visit.rs

```diff
@@ -904,7 +905,10 @@ fn noop_visit_fn_ret_ty<T: MutVisitor>(fn_ret_ty: &mut FnRetTy, vis: &mut T) {
 
 fn noop_visit_param_bound<T: MutVisitor>(pb: &mut GenericBound, vis: &mut T) {
     match pb {
-        GenericBound::Trait(ty, _modifier) => vis.visit_poly_trait_ref(ty),
+        GenericBound::Trait(ty, modifier) => {
+            vis.visit_poly_trait_ref(ty);
+            visit_trait_bound_modifier(modifier, vis);
```

Comment:
Could you visit things in the [standard order](https://github.com/rust-lang/rust/pull/126993) (e.g. nodes in source code order, spans last, etc)?